index.html

```diff
@@ -945,292 +656,31 @@ <h3>Extensions to the <code>Performance</code> Interface</h3>
-<a data-cite="HTML#or-equivalent" data-lt=
-'HTTP response codes equivalence'>equivalent</a>, then
-<ol>
-<li>If the current resource fails the <a>timing allow check</a>
-algorithm, set <a>redirectStart</a> and <a>redirectEnd</a> to 0.
-Then, return to the step labeled <a
-href="#dfn-step-fetch-start">Fetch start</a> with the redirected
-resource.</li>
-<li>If the value of redirectStart is not set, let it be the value
-of fetchStart.</li>
-<li>Let redirectEnd be the value of responseEnd.</li>
-<li>Set all the attributes in the <a>PerformanceResourceTiming</a>
-object to 0 except <a>startTime</a>, <a>redirectStart</a>,
-<a>redirectEnd</a>, and <a>initiatorType</a>.</li>
-<li>Return to the step labeled <a href="#dfn-step-fetch-start">Fetch
-start</a> with the redirected resource.</li>
-</ol>
-</li>
+<li>Create a <a>PerformanceResourceTiming</a> object, with its
+<a data-for="PerformanceResourceTiming">initiator type</a> set to |initiatorType|, its
+<a data-for="PerformanceResourceTiming">requested URL</a> set to |requestedURL|, and its
+<a data-for="PerformanceResourceTiming">timing info</a> set to |timingInfo|.
 <li><dfn data-lt="step-final-queue"><a data-cite=
 "PERFORMANCE-TIMELINE-2#dfn-queue-a-performanceentry">Queue</a> the
 <a>PerformanceResourceTiming</a> object</dfn>.</li>
 <li><a data-lt='add a PerformanceResourceTiming entry'>Add</a> the
-<a>PerformanceResourceTiming</a> object to the <a data-cite=
-"Fetch#concept-request">Request</a>'s <a>client</a>'s <a>global object</a>'s
+<a>PerformanceResourceTiming</a> object to |global|'s
 <a data-cite="PERFORMANCE-TIMELINE-2#dfn-performance-entry-buffer">performance
 entry buffer</a>.</li>
 </ol>
-<p class="issue">This specification does not specify whether steps
-20 and 21 should run before or after the <code>load</code> event of
-the resource—see <a href=
-"https://github.com/w3c/resource-timing/issues/82">issue 82</a> for
-related discussion.</p>
-<div class="issue" data-number="82"></div>
-</section>
-<section id="sec-monotonic-clock">
-<h3>Monotonic Clock</h3>
-<p>The value of the timing attributes MUST monotonically increase
-to ensure timing attributes are not skewed by adjustments to the
-system clock while <a data-cite="FETCH#concept-fetch" data-lt=
-'fetch'>fetching</a> the resource. The difference between any two
-chronologically recorded timing attributes MUST never be negative.
-For all resources, including subdocument resources, the user agent
-MUST record the system clock at the beginning of the root document
-navigation and define subsequent timing attributes in terms of a
-monotonic clock measuring time elapsed from the beginning of the
-navigation.</p>
-</section>
+</ol>
+<p>To <dfn>convert fetch timestamp</dfn> given <a>DOMHighResTimeStamp</a> |ts| and
+<a>global object</a> |global|, do the following:
+<ol>
+<li>If |ts| is zero, return zero.
+<li>Otherwise, return the <a>relative high resolution time</a> given |ts| and |global|.
```

Comment:
This now coarsens |ts| twice. I'm not sure if that's a problem?

Reply:
It's not. But I think it doesn't need to be re-coarsened. Let me check if it can be improved

Reply:
Yeah, coarsening twice is not ideal. Maybe we can expose the diff algorithm from the timestamps to the timeorigin as a separate algorithm?

Reply:
Why not. once https://github.com/w3c/hr-time/pull/114 is in I'll amend this one.